### DISCUSSION TITLE
Disable jump to anchor when using router.push()

### DISCUSSION THREAD

**Discussion by fiskolini** [vercel/next.js#24929] [Help]
Using next `10.2`, every time we push an URL with hash to the router, next is jumping to the anchor element.

    import {useRouter} from 'next/router'
    
    router.push('/contact#about-us');

This also happens if I disable `scroll`

    router.push('/contact#about-us', undefined, { scroll: false });

I didn't found any way to disable this behavior, nor a tweak to avoid it.
Any ideas?

**Comment by kylegoines** [comment]
i have also expirenced this same issue, the only fix i found was to completley circumvent next router and use `histroy.pushState`
my use case here
```
history.pushState('', '', `${router.asPath}#${current}`)
```

**Reply by fiskolini** [reply]
I managed to tweak it by adding this CSS
```css
html {
  scroll-behavior: smooth;
}
```

From an [answer](https://stackoverflow.com/a/67477942/2477831) to a question I raised on Stack Overflow
